packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/ios/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,523 @@
+        return
+    }
+
+    // If no remediations are present, abort the login process.
+    guard let remediation = currentResponse?.remediations.first else {
+        // Handle the error.
+        return
+    }
+
+    // Check for messages, such as entering an incorrect code.
+    if let message = response.messages.allMessages.first {
+        // Inform the view model to display the message.
+        return
+    }
+
+    // Handle the different sign-in steps (remediations) for a policy.
+    switch remediation.type {
+
+    // Username and password, though password may be separate; see the next case.
+    case .identify:
+        handleIdentify(remediation)
+
+        // Request a password or a passcode, such as one from Google Authenticator.
+        // Identity-first policies request the password separately.
+    case .challengeAuthenticator:
+        handleChallenge(remediation)
+
+        // Display a list of authenticators.
+    case .selectAuthenticatorEnroll:
+        handleSelectAuthenticatorToEnroll(remediation)
+
+    // Enroll in an Authenticator
+    case .enrollAuthenticator:
+        handleEnrollAuthenticator(remediation)
+
+    default:
+        // Handle the case of an unrecognized step.
+        // Note, there are several more types of remediations.
+    }
+}
+```
+
+The `handleIdentify(remediation:)` function updates the username and password fields in the remediation, and then calls `remediation.proceed()` which both sends the updated remediation to the server and requests the next step.
+
+
+```swift
+func handleIdentify(_ remediation: Remediation) {
+    // Update the values in the remediation object and go to the next step.
+    remediation["identifier"]?.value = username
+    remediation["credentials.passcode"]?.value = password
+    remediation.proceed()
+}
+```
+
+Some types of policies ask for the passsword in a separate step using a `.challengeAuthenticator` remediation type. The `handleChallege(remediation:)` function sets the password and calls `proceed()`:
+
+
+```swift
+// Sometimes the password is requested in a separate step.
+func handleChallenge(_ remediation: Remediation) {
+    guard let authenticator = remediation.authenticators.first else {
+        finish(with: .unexpectedAuthenticator)
+        return
+    }
+
+    let type = authenticator.type
+    switch type {
+    case .password:
+        remediation["credentials.passcode"]?.value = password
+        remediation.proceed()
+
+    default:
+        // Handle the case of an unknown authenticator.
+        return
+    }
+}
+
+```
+
+>**Note:** The response from the server specifies either the next step, an error, or a
+message, such as an incorrect code. The code that handles the response may need to show
+different UI, depending on the step. The `IDXClientDelegate` protocol includes two other functions: `idx(client:didReceive<Token>:)` handles receiving a sign-in token, and   `idx(client:didReceive<Error>:)` handles errors.
+
```

Comment:
Your note has some spurious line breaks in it. Can you delete them please?

Reply:
Nice Catch. Done.